### PR TITLE
Upgrade operator sdk to v1.5.0

### PR DESCRIPTION
**Pull request by amila-ku**
upgrade operator sdk to v1.5.0
update golang v1.16
added table driven tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,39 +1,142 @@
...
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/default | kubectl apply -f -
+
+undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build config/default | kubectl delete -f -
+
+
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+controller-gen: ## Download controller-gen locally if necessary.
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
```
Comment (pioneerit) [inline Makefile:113]: Just wondering, why do you switch from `tools.go` approach to directly call `go get ...`
Any reasons, or just own preferences?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Build the manager binary
+FROM golang:1.15 as builder
```
Comment (pioneerit) [inline Dockerfile]: You did upgrade for operator sdk to golang 1.16, do you want to update it here too?

Comment (pioneerit) [inline Dockerfile]: You did upgrade for operator sdk to golang 1.16, do you want to update it here too?

File: go.mod [hunk 3]
```diff
@@ -1,39 +1,14 @@
 module github.com/amila-ku/locust-operator
 
-go 1.13
+go 1.15
```
Comment (pioneerit) [inline go.mod]: The same question, if you want to use everywhere version 1.16

File: Makefile (hunk 1 above)
Comment (amila-ku) [reply Makefile:113]: I had my own makefile, this one uses the default provided by operator framework

File: Dockerfile (hunk 2 above)
Comment (amila-ku) [reply Dockerfile]: updated all ;-)

**Pull request merged**